indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,659 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import fetchRoomURL from 'indico-url:rooms_new.admin_room';
+import fetchRoomAttributesURL from 'indico-url:rooms_new.admin_room_attributes';
+import fetchAttributesURL from 'indico-url:rooms_new.admin_attributes';
+import fetchRoomAvailabilityURL from 'indico-url:rooms_new.admin_room_availability';
+import fetchRoomEquipmentURL from 'indico-url:rooms_new.admin_room_equipment';
+import updateRoomBasicDetailsURL from 'indico-url:rooms_new.admin_update_room';
+import updateRoomEquipmentURL from 'indico-url:rooms_new.admin_update_room_equipment';
+import updateRoomAttributesURL from 'indico-url:rooms_new.admin_update_room_attributes';
+import updateRoomAvailabilityURL from 'indico-url:rooms_new.admin_update_room_availability';
+import _ from 'lodash';
+import React from 'react';
+import {bindActionCreators} from 'redux';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {Button, Checkbox, Dimmer, Dropdown, Form, Grid, Header, Input, Loader, Message, Modal, TextArea} from 'semantic-ui-react';
+import {Form as FinalForm, Field} from 'react-final-form';
+import {FieldArray} from 'react-final-form-arrays';
+import arrayMutators from 'final-form-arrays';
+import shortid from 'shortid';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {snakifyKeys, camelizeKeys} from 'indico/utils/case';
+import {getChangedValues, ReduxCheckboxField, ReduxFormField} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+import PrincipalSearchField from 'indico/react/components/PrincipalSearchField';
+import EquipmentList from './EquipmentList';
+import DailyAvailability from './DailyAvailability';
+import NonBookablePeriods from './NonBookablePeriods';
+import SpriteImage from '../../components/SpriteImage';
+import {actions as roomsActions} from '../../common/rooms';
+import * as roomsSelectors from './selectors';
+import {selectors as userSelectors} from '../user';
+
+import './RoomEditModal.module.scss';
+
+
+function isInvalidNotificationPeriod(days) {
+    return days !== null && (days > 30 || days < 1);
+}
+
+function validate(fields) {
+    const {building, floor, number, capacity, attributes, surfaceArea, nonbookablePeriods, notificationBeforeDays,
+           notificationBeforeDaysWeekly, notificationBeforeDaysMonthly} = fields;
+    const errors = {};
+    if (!building) {
+        errors.building = Translate.string('Please provide a building.');
+    }
+    if (!floor) {
+        errors.floor = Translate.string('Please provide a floor.');
+    }
+    if (!number) {
+        errors.number = Translate.string('Please provide a number.');
+    }
+    if (!capacity) {
```

Comment:
Do we need those two `if`s? Can't we just use the second one? I don't think it is that necessary to have different messages for a missing and invalid value.

Reply:
The first message is for the missing room number (ie. 31/01-19) and the second one is for the capacity of the room.

Reply:
Yes, of course. However I had in mind lines 70 and 73.